pkg/utils/validation/kubernetesversion/version.go

```diff
@@ -20,8 +23,18 @@ var SupportedVersions = []string{
 	"1.34",
 }
 
+// EXPERIMENTAL_DISABLE_VERSION_CHECK_ENV holds the name of the environment variable to prevent a crash
+// if the detected k8s version is not in the list of supported k8s versions.
+// This should only be used if you know exactly what you are doing and on your own risk.
+const EXPERIMENTAL_DISABLE_VERSION_CHECK_ENV = "DO_NOT_CRASH_ON_UNSUPPORTED_KUBERNETES_VERSION"
```

Comment:
The idiomatic naming pattern for consts is to use mixed caps.

From [Effective Go > MixedCaps](https://go.dev/doc/effective_go#mixed-caps):
> Finally, the convention in Go is to use MixedCaps or mixedCaps rather than underscores to write multiword names.


```suggestion
const ExperimentalDisableVersionCheckEnv = "DO_NOT_CRASH_ON_UNSUPPORTED_KUBERNETES_VERSION"
```

Reply:
Done via: [`af5e8f3` (#13221)](https://github.com/gardener/gardener/pull/13221/commits/af5e8f3a7b80482269bbd11eaed2c7a439d455e7)